src/Sylius/Bundle/ApiBundle/Serializer/PromotionDenormalizer.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Sylius\Bundle\ApiBundle\Serializer;
+
+use Sylius\Bundle\ApiBundle\Helpers\PromotionPriceTransformer;
+use Symfony\Component\Serializer\Normalizer\ContextAwareDenormalizerInterface;
+use Sylius\Component\Core\Model\Promotion;
+use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
+
+class PromotionDenormalizer implements ContextAwareDenormalizerInterface
+{
+    private const ALREADY_CALLED = 'promotion_denormalizer_already_called';
+    /**
+     * @var DenormalizerInterface
+     */
+    private DenormalizerInterface $objectNormalizer;
+    private string $classType;
+
+    public function __construct(
+        DenormalizerInterface $objectNormalizer,
+        string $classType
+    )
+    {
+        $this->objectNormalizer = $objectNormalizer;
+        $this->classType = $classType;
+    }
+
+    public function denormalize($data, string $type, string $format = null, array $context = [])
+    {
+        $channelCode = $context[ContextKeys::CHANNEL]->getCode();
+        $context['api_allow_update'] = true;
```

Comment:
What's the point to allow update?

Reply:
because of this: https://prnt.sc/1w81yd1